sentry-android-core/src/main/java/io/sentry/android/core/gestures/SentryGestureListener.java

```diff
@@ -0,0 +1,238 @@
+package io.sentry.android.core.gestures;
+
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.Window;
+import io.sentry.Breadcrumb;
+import io.sentry.IHub;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.SentryAndroidOptions;
+import java.lang.ref.WeakReference;
+import java.util.Collections;
+import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class SentryGestureListener implements GestureDetector.OnGestureListener {
+
+  private final @NotNull WeakReference<Window> windowRef;
+  private final @NotNull IHub hub;
+  private final @NotNull SentryAndroidOptions options;
+
+  private final ScrollState scrollState = new ScrollState();
+
+  public SentryGestureListener(
+    final @NotNull WeakReference<Window> windowRef,
+    final @NotNull IHub hub,
+    final @NotNull SentryAndroidOptions options
+  ) {
+    this.windowRef = windowRef;
+    this.hub = hub;
+    this.options = options;
+  }
+
+  public void onUp(final @NotNull MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onUp");
+    final View scrollTarget = scrollState.targetRef.get();
+    if (decorView == null || scrollTarget == null) {
+      return;
+    }
+
+    if (scrollState.type == null) {
+      options
+        .getLogger()
+        .log(SentryLevel.INFO, "Unable to define scroll type. No breadcrumb captured.");
```

Comment:
Is this worth an INFO level or it's only useful if we're debugging something (why didn't the crumb got added?)
```suggestion
        .log(SentryLevel.DEBUG, "Unable to define scroll type. No breadcrumb captured.");
```